react/features/stream-effects/screenshot-capture/ScreenshotCaptureEffect.js

```diff
@@ -50,23 +50,18 @@ export default class ScreenshotCaptureEffect {
         this._streamWorker.onmessage = this._handleWorkerAction;
     }
 
-    /**
-     * Checks if the local track supports this effect.
-     *
-     * @param {JitsiLocalTrack} jitsiLocalTrack - Targeted local track.
-     * @returns {boolean} - Returns true if this effect can run on the specified track, false otherwise.
-     */
-    isEnabled(jitsiLocalTrack: Object) {
-        return jitsiLocalTrack.isVideoTrack() && jitsiLocalTrack.videoType === 'desktop';
-    }
-
     /**
      * Starts the screenshot capture event on a loop.
      *
      * @param {MediaStream} stream - The desktop stream from which screenshots are to be sent.
+     * @param {string} videoType - The type of the media stream.
      * @returns {MediaStream} - The same stream, with the interval set.
      */
-    startEffect(stream: MediaStream) {
+    startEffect(stream: MediaStream, videoType: string) {
+        if (videoType !== 'desktop') {
+            return;
+        }
```

Comment:
Can we return a Promise here instead of the stream since we don't use the returned stream ? Promise should be rejected when videotype validation fails.